extensions/persistence.md

```diff
@@ -0,0 +1,160 @@
+
+If persistence is enabled, the server MUST send a `JOIN` line for each channel the client is joined to, subsequent to sending either `376` or `422`.
+
+### `PERSISTENCE GET`
+
+The `GET` subcommand is sent from the client to the server:
+
+    PERSISTENCE GET
+
+The server replies with a `PERSISTENCE STATUS` command as specified above.
+
+### `PERSISTENCE SET`
+
+The `SET` subcommand is sent from the client to the server:
+
+    PERSISTENCE SET <client-setting>
+
+where `<client-setting>` takes one of the three values specified above; the server replies with a `PERSISTENCE STATUS` command as specified above.
+
+#### Errors and Warnings
+
+Servers MUST NOT allow unauthenticated clients to become persistent, since there is no way for the client to reconnect in that case. Unauthenticated clients attempting to issue `GET` or `SET` MUST receive:
+
+    FAIL PERSISTENCE ACCOUNT_REQUIRED :An account is required
+
+On invalid parameters, the server MUST return:
+
+    FAIL PERSISTENCE INVALID_PARAMETERS :Invalid parameters
+
+If persistence cannot be accessed or modified due to an error, the `INTERNAL_ERROR` error code SHOULD be returned.
+
+    FAIL PERSISTENCE INTERNAL_ERROR :An error occurred
+
+Finally, servers MAY accept the change in client setting while not changing the effective setting. For example, persistence may be mandatory for all users, or it may be reserved for privileged users. In this event, the server MUST NOT send a `FAIL` response, but MUST send a `PERSISTENCE STATUS` response describing the current settings.
+
+### Examples
+
+A client enables persistence on a server with opt-in persistence:
+~~~~
+[c] PERSISTENCE GET
+[s] PERSISTENCE STATUS DEFAULT OFF
+[c] PERSISTENCE SET ON
+[s] PERSISTENCE STATUS ON ON
+~~~~
+
+A client attempts to enable persistence, but server policy does not allow them to become persistent:
+
+~~~~
+[c] PERSISTENCE SET ON
+[s] PERSISTENCE STATUS ON OFF
+~~~~
+
+A client attempts to disable persistence, but server policy requires that they remain persistent:
+
+~~~~
+[c] PERSISTENCE GET
+[s] PERSISTENCE STATUS DEFAULT ON
+[c] PERSISTENCE SET OFF
+[s] PERSISTENCE STATUS OFF ON
+~~~~
+
+A client negotiates the `draft/persistence` capability and sees that persistence is enabled for them:
+
+~~~~
+[c] CAP REQ :draft/persistence sasl
+[s] :irc.example.com CAP * ACK :draft/persistence sasl
+[c] NICK testuser
+[c] USER u s e r
+[c] AUTHENTICATE PLAIN
+[c] AUTHENTICATE amlsbGVzAGppbGxlcwBzZXNhbWU=
+[c] CAP END
+[s] AUTHENTICATE +
+[s] :irc.example.com 900 * * jilles :You are now logged in as jilles
+[s] :irc.example.com 903 * :Authentication successful
```

Comment:
Could you reorder SASL messages to be in the "canonical" order?

Reply:
This was intentional, but yeah, it was unnecessarily edgy :-)